### PR TITLE
Remove NetNative uapaot support

### PR DESCRIPTION
**Pull request by ViktorHofer**
Reacts to https://github.com/dotnet/corefx/pull/38113#issuecomment-498253778 to unblock nullability work.

@jkotas I left the AOT source optimizations as we might need them for MonoAOT? If not I'm happy to send another commit with the source removals.

cc @danmosemsft @ericstj @safern

### CODE REVIEW

**Comment by jkotas** [comment]
> AOT source optimizations as we might need them for MonoAOT

The true optimizations for AOT or small footprint, like `*.SizeOpt.cs` files, should stay. They are likely going to be useful in future as you have said.

It should be fine to delete `dir /s *.CoreRT.cs` and `dir /s *.uapaot.txt`.

**Comment by jkotas** [comment]
Also, it should be fine to delete `#if FEATURE_SERIALIZATION_UAPAOT / #if !FEATURE_SERIALIZATION_UAPAOT` or `#if uapaot / #if !uapaot` - probably in a separate change.

**Comment by danmosemsft** [comment]
>Also, it should be fine to delete #if ...

If you clone and build https://github.com/dotnet/codeformatter then run eg `...\codeformatter\bin\DeadRegions\Debug\DeadRegions.exe some.csproj /disable FOO /edit` will remove all regions active only when FOO is defined (and there are other variations)

**Comment by danmosemsft** [comment]
> It should be fine to delete dir /s *.CoreRT.cs and dir /s *.uapaot.txt.

What about the rd.xml? I assume we keep ILLinkTrim.xml

**Comment by jkotas** [comment]
> What about the rd.xml?

Yes, `*rd.xml` should be ok to delete too.

**Comment by jkotas** [comment]
Are there any cases where this cleanup affects standalone packages and where it would break newer version of those packages for .NET Native? Something to watch for.

**Comment by ViktorHofer** [comment]
> Are there any cases where this cleanup affects standalone packages and where it would break newer version of those packages for .NET Native? Something to watch for.

Yes, that's a very good point. I need to harvest 3-4 package assets for uapaot.

**Comment by ericstj** [comment]
These might be more fodder for deletion:
```
shims\ApiCompatBaseline.uapaot.netstandard.txt
shims\ApiCompatBaseline.uapaot.netstandardOnly.txt
System.Data.Common\src\MatchingRefApiCompatBaseline.uapaot.txt
System.Memory\src\ApiCompatBaseline.uapaot.txt
System.Reflection\src\ApiCompatBaseline.uapaot.txt
System.Reflection.Primitives\src\ApiCompatBaseline.uapaot.txt
System.Runtime\src\ApiCompatBaseline.uapaot.txt
System.Runtime.Extensions\src\ApiCompatBaseline.uapaot.txt
System.Runtime.InteropServices\src\ApiCompatBaseline.uapaot.txt
System.Runtime.Loader\src\ApiCompatBaseline.uapaot.txt
System.Security.Cryptography.X509Certificates\tests\DSAOther.cs
System.Security.Cryptography.X509Certificates\tests\ECDsaOther.cs
System.Security.Cryptography.X509Certificates\tests\RSAOther.cs
System.Threading\src\ApiCompatBaseline.uapaot.txt
System.Threading.ThreadPool\src\ApiCompatBaseline.uapaot.txt
System.Xml.XmlSerializer\src\MatchingRefApiCompatBaseline.uapaot.txt
```

**Comment by ViktorHofer** [comment]
I need to harvest the preview packages here (until we ship .NET Core 3, afterwards we can simply retarget to the stable ones) otherwise I don't have a way to stop building uapaot here without getting compile version runtime version conflicts. Unfortunately it seems this is blocked atm. I need to talk with @ericstj about this.

**Comment by bartonjs** [comment]
@ericstj How'd you determine these files?

System.Security.Cryptography.X509Certificates\tests\DSAOther.cs
System.Security.Cryptography.X509Certificates\tests\ECDsaOther.cs
System.Security.Cryptography.X509Certificates\tests\RSAOther.cs

(They're a "make a 3rd party type so we understand what happens if someone extends our non-sealed classes" portion of a test that is in use in netcoreapp)

**Comment by ViktorHofer** [comment]
Status: I need to fix the packaging for System.Reflection.Emit and then this PR should be ready.

**Comment by stephentoub** [comment]
@viktorhofer, will this go in today? Thanks.

**Comment by ViktorHofer** [comment]
Yes this should be ready later today.

File: pkg/test/project.csproj.template [hunk 1]
```diff
@@ -10,6 +10,8 @@
   </PropertyGroup>
   
   <ItemGroup>
+    <!-- System.Reflection.Emit.Lightweight version has been bumped so we expect validation to fail until we ingest the new core-setup which contains that change -->
+    <IgnoredReference Include="System.Reflection.Emit.Lightweight" Version="4.1.0.0" />
```
Comment (ericstj) [inline pkg/test/project.csproj.template]: I believe there were two assemblies that hit this?

**Comment by ViktorHofer** [comment]
I'll submit a follow-up PR for cleaning up the discussed files.

**Pull request merged**